components/cdc/src/observer.rs

```diff
@@ -258,23 +258,6 @@ impl<S: EngineSnapshot> OldValueReader<S> {
             .map(|v| v.deref().to_vec())
     }
 
-    fn check_lock(&mut self, key: &Key, statistics: &mut Statistics) -> bool {
-        statistics.lock.get += 1;
-        let mut opts = ReadOptions::new();
-        opts.set_fill_cache(false);
-        let key_slice = key.as_encoded();
-        let user_key = Key::from_encoded_slice(Key::truncate_ts_for(key_slice).unwrap());
-
-        match self.snapshot.get_cf_opt(opts, CF_LOCK, &user_key).unwrap() {
-            Some(v) => {
-                let lock = Lock::parse(v.deref()).unwrap();
-                std::cmp::max(lock.ts, lock.for_update_ts)
-                    == Key::decode_ts_from(key_slice).unwrap()
-            }
-            None => false,
-        }
-    }
-
     // return Some(vec![]) if value is empty.
     // return None if key not exist.
```

Comment:
I'm not clear why there is such an explicit comment for it... But now it looks valid to me?

Reply:
🤔emmm, Isn't it that all `Some(vec![])` cases are replaced with `None`s?

Reply:
I think it was wrong previously. @5kbpers Could you take a look?

Reply:
Previously, `Some(vec![])` and `None` have different meanings. I thought this PR intentionally  removed one of them before..
Ah... and by the way.. looks like the comment for `fn get_value_default` is also not very correct... @5kbpers PTAL..

Reply:
Should remove these redundant comments.

Reply:
It was used for detecting whether the value exists (If not exist, there must be some expectations), now we don't need this since we can ensure the value exist (previously the key could be removed by region migration or other events)